app/src/main/java/org/mozilla/focus/browser/integration/BrowserToolbarIntegration.kt

```diff
@@ -150,12 +159,35 @@ class BrowserToolbarIntegration(
 
         customTabsFeature?.start()
         navigationButtonsIntegration?.start()
+        observerSecurityIndicatorChanges()
+    }
+
+    private fun observerSecurityIndicatorChanges() {
+        securityIndicatorScope = store.flowScoped { flow ->
+            flow.mapNotNull { state ->
+                state.findCustomTabOrSelectedTab(customTabId)
+            }.ifChanged { tab -> tab.content.securityInfo }
+                .collect {
+                    if (!it.content.securityInfo.secure) {
+                        if (Indicators.SECURITY !in toolbar.display.indicators) {
+                            toolbar.display.apply {
+                                indicators = indicators + Indicators.SECURITY
+                            }
+                        }
+                    } else if (Indicators.SECURITY in toolbar.display.indicators && it.content.securityInfo.secure) {
+                        toolbar.display.apply {
+                            indicators = indicators - Indicators.SECURITY
+                        }
+                    }
+                }
+        }
```

Comment:
nit: There is an opportunity to write this more compact:
- Since we only need the boolean flag, we can map to that flag and then use `ifChanged()`
- On the indicator list we can use `+=` and `-=" which avoids using those multi-line `apply()` statements
- Currently one of the if cases is nested and the other is not

So we end up with something like this:

```Kotlin
        securityIndicatorScope = store.flowScoped { flow ->
            flow
                .mapNotNull { state -> state.findCustomTabOrSelectedTab(customTabId)?.content?.securityInfo?.secure }
                .ifChanged()
                .collect { secure ->
                    if (secure && Indicators.SECURITY in toolbar.display.indicators) {
                        toolbar.display.indicators -= Indicators.SECURITY
                    } else if (!secure && Indicators.SECURITY !in toolbar.display.indicators) {
                        toolbar.display.indicators += Indicators.SECURITY
                    }
                }
        }
```

Reply:
✨